hashmap.js

```diff
@@ -25,7 +25,14 @@
 		this.clear();
 		switch (arguments.length) {
 			case 0: break;
-			case 1: this.copy(other); break;
+			case 1: {
+				if(Array.isArray(other)) {
+					this.from2DArray(other);

```

Comment:
You have the `multi()` function for this. You can delete `from2DArray` and change this for `multi(this, other);`